app/viewmodels/authorisedOfficials/AuthorisedOfficialsStatusHelper.scala

```diff
@@ -0,0 +1,102 @@
+import viewmodels.StatusHelper
+
+object AuthorisedOfficialsStatusHelper extends StatusHelper {
+
+  private def journeyCommon(index:Int): Seq[QuestionPage[_]] = Seq(
+    AuthorisedOfficialsNamePage(index),
+    AuthorisedOfficialsDOBPage(index),
+    AuthorisedOfficialsPhoneNumberPage(index),
+    AuthorisedOfficialsPositionPage(index),
+    IsAuthorisedOfficialNinoPage(index),
+    AuthorisedOfficialAddressLookupPage(index),
+    IsAuthorisedOfficialPreviousAddressPage(index)
+  )
+
+  private def remainingJourneyPages(index:Int): Seq[QuestionPage[_]] = Seq(
+    AuthorisedOfficialsNinoPage(index),
+    AuthorisedOfficialsPassportPage(index),
+    AuthorisedOfficialPreviousAddressLookupPage(index)
+  )
+
+  private val authorisedOfficial1common: Seq[QuestionPage[_]] = journeyCommon(0) ++ Seq(IsAddAnotherAuthorisedOfficialPage)
+
+  private val authorisedOfficial2common: Seq[QuestionPage[_]] = journeyCommon(1)
+
+  private val allPages: Seq[QuestionPage[_]] =  {
+    authorisedOfficial1common ++ authorisedOfficial2common ++
+      remainingJourneyPages(0) ++ remainingJourneyPages(1)
+  }
+
+  private val f1 = (isNino:Boolean) => {
+    if(isNino) {
+      authorisedOfficial1common ++ Seq(AuthorisedOfficialsNinoPage(0))
+    } else {
+      authorisedOfficial1common ++ Seq(AuthorisedOfficialsPassportPage(0))
+    }
+  }
+
+  private val f2 = (list: Seq[QuestionPage[_]], isPreviousAddress:Boolean) => {
+    if(isPreviousAddress) list ++ Seq(AuthorisedOfficialPreviousAddressLookupPage(0)) else list
+  }
+
+  private val f3 = (list: Seq[QuestionPage[_]], isNino:Boolean) => {
+    if(isNino) {
+      list ++ authorisedOfficial2common ++ Seq(AuthorisedOfficialsNinoPage(1))
+    } else {
+      list ++ authorisedOfficial2common ++ Seq(AuthorisedOfficialsPassportPage(1))
+    }
+  }
+
+  private val f4 = (list: Seq[QuestionPage[_]], isPreviousAddress:Boolean) => {
+    if(isPreviousAddress) list ++ Seq(AuthorisedOfficialPreviousAddressLookupPage(1)) else list
+  }
+
+
+  override def checkComplete(userAnswers: UserAnswers): Boolean = {
+
+    def noAdditionalPagesDefined(list: Seq[QuestionPage[_]]): Boolean = userAnswers.unneededPagesNotPresent(list, allPages)
+
+    (userAnswers.get(IsAuthorisedOfficialNinoPage(0)), userAnswers.get(IsAuthorisedOfficialPreviousAddressPage(0))) match {
+      case (Some(isNino1), Some(isPreviousAddress1)) =>
+
+        userAnswers.get(IsAddAnotherAuthorisedOfficialPage) match {
+          case Some(false) =>
+            val newPages = f2(f1(isNino1), isPreviousAddress1)
+            userAnswers.arePagesDefined(newPages) && noAdditionalPagesDefined(newPages)
+          case Some(true) =>
+
+            (userAnswers.get(IsAuthorisedOfficialNinoPage(1)), userAnswers.get(IsAuthorisedOfficialPreviousAddressPage(1)))match {
+              case (Some(isNino2), Some(isPreviousAddress2)) =>
+                val newPages = f4(f3(f2(f1(isNino1), isPreviousAddress1), isNino2), isPreviousAddress2)
```

Comment:
this would be much cleaner and more readable if we implemented an implicit class here, like:

`isPreviousAddress1.f1(isNino1).f2(isPreviousAddress1).f3(isNino2).f4(isPreviousAddress2)`

(with clearer names, if possible)